### PR TITLE
Implement AddNoteCommand

### LINKED ISSUE #143: Add Notes to Memo
As a tutor who wants to ensure the smoothest teaching process, I can add notes to my memo without having to switch to the memo tab so that I can remain other tabs while taking down notes.

### PR DESCRIPTION
**Pull request by nweiyue**
- `addnote NOTE` appends NOTE on a new line to the end of memo
-  Auto-saves memo data when exiting

### CODE REVIEW

**Review by marcustw — COMMENTED** [review]
LGTM so far!

File: src/test/java/atas/logic/commands/memo/AddNoteCommandTest.java [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+        }
+
+        @Override
+        public ObservableList<Student> getFilteredStudentList() {
+            throw new AssertionError("This method should not be called.");
+        }
+
+        @Override
+        public ObservableList<Session> getFilteredSessionList() {
+            return null;
```
Comment (marcustw) [inline src/test/java/atas/logic/commands/memo/AddNoteCommandTest.java]: Perhaps throw an AssertionError here instead?

**Review by marcustw — APPROVED** [review]
LGTM! Here are some nitpicks by me.

File: src/test/java/atas/logic/commands/memo/AddNoteCommandTest.java (hunk 1 above)
Comment (nweiyue) [reply src/test/java/atas/logic/commands/memo/AddNoteCommandTest.java]: Good suggestion! Will make the necessary changes 👍

**Pull request merged**